include/Drivers/MotorController.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef MOTORCONTROLLER_HPP
+#define MOTORCONTROLLER_HPP
+
+#include <Arduino.h>
+#include "RoboClaw.h"
+
+namespace motors {
+
+enum MotorMode { TORQUE, SPEED, VELOCITY };
+
+class MotorController {
+public:
+  MotorController(){};
+  virtual void begin(long speed);
+  // TODO: virtual void setMode(uint8_t channel, MotorMode mode) = 0;
+  virtual int32_t getSpeed(uint8_t address, uint8_t channel, bool &valid) = 0;
+  virtual void setSpeed(int32_t speed, uint8_t address, uint8_t channel, bool &valid) = 0;
+  // TODO: virtual int32_t getPosition(uint8_t channel) = 0;
+  // TODO: virtual void setPosition(uint8_t channel) = 0;
+};
+
+class RoboClawController : public MotorController {
```

Comment:
Once again I'd probably separate these classes into different files. It makes your architecture a little hard to follow unless you know it well